src/CoreBundle/private/library/classes/TCMSURLHistory.class.php

```diff
@@ -168,21 +184,22 @@ public function EncodeParameters($aParameters, $returnWithFragment = true)
      */
     public function Clear($id)
     {
-        $this->index = $id;
-        $endpoint = count($this->aHistory)-1;
+        $endpoint = count($this->aHistory) - 1;
         if ($endpoint > 0) {
             for ($i = $endpoint; $i > $id; --$i) {
                 unset($this->aHistory[$i]);
             }
         }
+
+        // reset the keys to prevent key gaps
+        $this->aHistory = array_values($this->aHistory);
```

Comment:
I'd suspect we can also remove this one as we only remove items from the end.

Reply:
Makes me nervous, but i removed it. I had issues with a wrong index, but i have no idea, how this happened.